src/piv.rs

```diff
@@ -339,8 +350,70 @@ impl RetiredSlotId {
     }
 }
 
+/// Management slot IDs.
+#[allow(missing_docs)]
+#[derive(Clone, Copy, Debug, PartialEq)]
+pub enum ManagementSlotId {
+    PIN,
+    PUK,
+    Management,
+}
+
+impl TryFrom<u8> for ManagementSlotId {
+    type Error = Error;
+
+    fn try_from(value: u8) -> Result<Self> {
+        match value {
+            0x80 => Ok(ManagementSlotId::PIN),
+            0x81 => Ok(ManagementSlotId::PUK),
+            0x9b => Ok(ManagementSlotId::Management),
+            _ => Err(Error::InvalidObject),
+        }
+    }
+}
+
+impl FromStr for ManagementSlotId {
+    type Err = Error;
+
+    fn from_str(value: &str) -> Result<Self> {
+        match value {
+            "80" => Ok(ManagementSlotId::PIN),
+            "81" => Ok(ManagementSlotId::PUK),
+            "9b" => Ok(ManagementSlotId::Management),
+            _ => Err(Error::InvalidObject),
+        }
+    }
+}
+
+impl From<ManagementSlotId> for u8 {
+    fn from(slot: ManagementSlotId) -> u8 {
+        match slot {
+            ManagementSlotId::PIN => 0x80,
+            ManagementSlotId::PUK => 0x81,
+            ManagementSlotId::Management => 0x9b,
+        }
+    }
+}
+
+impl Display for ManagementSlotId {
+    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
+        write!(f, "{:?}", self)
+    }
+}
+
+impl ManagementSlotId {
+    /// Returns the [`ObjectId`] that corresponds to a given [`ManagementSlotId`].
+    pub(crate) fn object_id(self) -> ObjectId {
+        match self {
+            ManagementSlotId::PIN => 0x005f_c10b, // TODO: no idea where to get those object_ids
```

Comment:
I'd love to get help on this one, I do not know where those values come from.

Reply:
See [NIST SP 800-73-4](https://csrc.nist.gov/publications/detail/sp/800-73/4/final) Table 3: Object Identifiers of the PIV Data Objects for Interoperable Use

> Data Object: X.509 Certificate for Key Management
> OID: 2.16.840.1.101.3.7.2.1.2
> BER-TLV tag: '5FC10B'